src/main/java/servicii/AccountService.java

```diff
@@ -0,0 +1,93 @@
+package servicii;
+
+import cache.AccountList;
+import cache.DataBase;
+import cache.UserList;
+import utilizatori.Client;
+
+public class AccountService {
+
+    private final DataBase<UserList> userListDataBase = new DataBase<UserList>("userList.dat");
+    private final DataBase<AccountList> accountListDataBase = new DataBase<AccountList>("accountList.dat");
+    private UserList userList;
+    private AccountList accountList;
+
+    public void moneyTransfer(String fromIban, String toIban, double amount){
+        this.userList = userListDataBase.readObject();
+        this.accountList = accountListDataBase.readObject();
+
+        Client fromClient = userList.read(accountList.getUser(fromIban));
+        Client toClient = userList.read(accountList.getUser(toIban));
+
+        decreaseAccountAmountDebit(fromIban, amount, fromClient);
+        toClient.getDebitAccountByIban(toIban).increaseAmountBy(amount);
+        System.out.println("Transfer successful!");
+
+        this.userList.update(fromClient.getUserName(), fromClient);
+        this.userList.update(toClient.getUserName(), toClient);
+        this.userListDataBase.writeObject(this.userList);
+
+    }
+
+    public void moneyExtraction(String fromIban, double amount){
+        this.userList = userListDataBase.readObject();
+        this.accountList = accountListDataBase.readObject();
+
+        Client fromClient = userList.read(accountList.getUser(fromIban));
+        decreaseAccountAmountDebit(fromIban,amount, fromClient);
+        System.out.println("Payment successful!");
+
+        this.userList.update(fromClient.getUserName(), fromClient);
+        this.userListDataBase.writeObject(this.userList);
+    }
+
+    public void depositToAccount(String toIban, double amount){
+        this.userList = userListDataBase.readObject();
+        this.accountList = accountListDataBase.readObject();
+
+        Client toClient = userList.read(accountList.getUser(toIban));
+        toClient.getDebitAccountByIban(toIban).increaseAmountBy(amount);
+        this.userList.update(toClient.getUserName(), toClient);
+        this.userListDataBase.writeObject(this.userList);
+    }
+
+    public void creditExtraction(String fromIban, double amount){
+        this.userList = userListDataBase.readObject();
+        this.accountList = accountListDataBase.readObject();
+
+        Client fromClient = userList.read(accountList.getUser(fromIban));
+        decreaseAccountAmountDebit(fromIban,amount, fromClient);
+        System.out.println("Payment successful!");
+
+        this.userList.update(fromClient.getUserName(), fromClient);
+        this.userListDataBase.writeObject(this.userList);
+    }
+
+    private  void decreaseAccountAmountCredit(String fromIban, double amount, Client fromClient){
+        if(hasCreditAvailable(amount,fromClient.getCreditAccountById(fromIban).getLimitAmount(), fromClient.getCreditAccountById(fromIban).getUsedAmount())){

```

Comment:
This line seems a bit difficult to read. Maybe it would be better to first initialize the variables which will then be used in the hasCreditAvailable(...) method.